pkg/sql/sem/builtins/pg_builtins.go

```diff
@@ -1065,70 +1065,70 @@ var pgBuiltins = map[string]builtinDefinition{
-				}
-				if tabOID := *args[0].(*tree.DOid); tabOID.Oid >= catconstants.MinVirtualID {
-					// Virtual table columns do not have descriptions.
-					return tree.DNull, nil
-				}
-
-				// Note: the following is equivalent to:
-				//
-				// SELECT description FROM pg_catalog.pg_description
-				//  WHERE objoid=$1 AND objsubid=$2 LIMIT 1
-				//
-				// TODO(jordanlewis): Really we'd like to query this directly
-				// on pg_description and let predicate push-down do its job.
-				r, err := evalCtx.Planner.QueryRowEx(
-					ctx, "pg_get_coldesc",
-					sessiondata.NoSessionDataOverride,
-					`
-SELECT comment FROM system.comments c
-WHERE c.type=$1::int AND c.object_id=$2::int AND c.sub_id=$3::int LIMIT 1
-`, catalogkeys.ColumnCommentType, args[0], args[1])
-				if err != nil {
-					return nil, err
-				}
-				if len(r) == 0 {
-					return tree.DNull, nil
-				}
-				return r[0], nil
-			},
+			// Note: the following is equivalent to:
+			//
+			// SELECT description FROM pg_catalog.pg_description
+			//  WHERE objoid=$1 AND objsubid=$2 LIMIT 1
+			//
+			// TODO(jordanlewis): Really we'd like to query this directly
+			// on pg_description and let predicate push-down do its job.
+			Body: fmt.Sprintf(
+				`SELECT comment
+         FROM system.comments c
+				 WHERE c.type=%[1]d
+				 AND c.object_id=$1::int
+				 AND c.sub_id=$2::int
+				 AND $1 < %[2]d /* Virtual table columns do not have descriptions. */
+				 AND $2 != 0 /* Column ID 0 never exists, and we don't want the query
+					              to pick up the table comment by accident. */
+				 LIMIT 1`,
+				catalogkeys.ColumnCommentType,
+				catconstants.MinVirtualID),
 			Info:       notUsableInfo,
 			Volatility: volatility.Stable,
 		},
 	),
 
 	"obj_description": makeBuiltin(defProps(),
 		tree.Overload{
+			IsUDF:      true,
 			Types:      tree.ParamTypes{{Name: "object_oid", Typ: types.Oid}},
 			ReturnType: tree.FixedReturnType(types.String),
-			Fn: func(ctx context.Context, evalCtx *eval.Context, args tree.Datums) (tree.Datum, error) {
-				return getPgObjDesc(ctx, evalCtx, "", args[0].(*tree.DOid).Oid)
-			},
+			Body: `SELECT description
+						 FROM pg_catalog.pg_description
+						 WHERE objoid = $1
+						 AND objsubid = 0
+						 LIMIT 1`,
 			Info:       notUsableInfo,
 			Volatility: volatility.Stable,
 		},
 		tree.Overload{
+			IsUDF:      true,
 			Types:      tree.ParamTypes{{Name: "object_oid", Typ: types.Oid}, {Name: "catalog_name", Typ: types.String}},
 			ReturnType: tree.FixedReturnType(types.String),
-			Fn: func(ctx context.Context, evalCtx *eval.Context, args tree.Datums) (tree.Datum, error) {
-				return getPgObjDesc(
-					ctx,
-					evalCtx,
-					string(tree.MustBeDString(args[1])),
-					args[0].(*tree.DOid).Oid,
-				)
-			},
+			Body: fmt.Sprintf(
+				`SELECT description
+				 FROM pg_catalog.pg_description
+				 WHERE objoid = $1
+				 AND objsubid = 0
+				 AND classoid = CASE $2
+					 WHEN 'pg_class' THEN %[1]d
```

Comment:
I think we should do this using a join on pg_class rather than a hardcoded case/when list, unless there's a strong reason to do it this way? I guess it's maybe more efficient to do it this way - but will there be any compat issues with this?

Reply:
what did you have in mind? do you mean `AND classoid = $2::regclass::oid`? i think that will have a performance penalty, but i'll check.

Reply:
Here's how I prototyped it. It definitely might be slower.

https://github.com/jordanlewis/cockroach/blob/show-functions-and-udf-builtins/pkg/sql/sem/builtins/pg_builtins.go#L1010-L1023

Reply:
the roundtrips were not affected by the JOIN approach, nor was the latency in my local manual testing, so i'll do it that way